modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -470,11 +484,23 @@ func (fsrv *FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request, next c
 		w = statusOverrideResponseWriter{ResponseWriter: w, code: statusCodeOverride}
 	}
 
+	// Nix uses modtime=1 for files in its store, and it's useless to set
+	// the Last-Modified header in that case, so we need to check if the
+	// header was set by anything else first (i.e. user config or another
+	// module) so that we can later check if http.ServeContent set the header.
+	modTime := info.ModTime()
+	beforeLastModified := w.Header().Get("Last-Modified")
+
 	// let the standard library do what it does best; note, however,
 	// that errors generated by ServeContent are written immediately
 	// to the response, so we cannot handle them (but errors there
 	// are rare)
-	http.ServeContent(w, r, info.Name(), info.ModTime(), file.(io.ReadSeeker))
+	http.ServeContent(w, r, info.Name(), modTime, file.(io.ReadSeeker))
+
+	// Delete the Last-Modified header if it was changed and modtime=1 (i.e. Nix store)
+	if beforeLastModified != w.Header().Get("Last-Modified") && modTime.Unix() == 1 {
+		w.Header().Del("Last-Modified")
```

Comment:
Does this work? When `http.ServeContent()` returns, the response is already written.

Reply:
Yeah, good point, I'm not sure. I didn't try it.

Reply:
This is something we could handle with the same ResponseWriterWrapper trick, right? So maybe the mtime=1 special casing doesn't need to be upstream at all.

Reply:
Can you just move these lines up to before ServeContent is called?

Reply:
No because `ServeContent` is what is setting `Last-Modified`. That's what we're trying to intercept in this case.

Reply:
Oh right, duh.

Yeah, a ResponseWriter wrapper is probably best then.